scripts/sipify.py

```diff
@@ -0,0 +1,2853 @@
+            prefix, arguments, suffix = signature_split.groups()
+            suffix = ')' + suffix
+        else:
+            prefix = ''
+            arguments = input_string
+            suffix = ''
+
+    arguments_list = split_args(arguments)
+
+    if CONTEXT.is_qt6:
+        filtered_args = [arg for arg in arguments_list if
+                         'SIP_PYARGREMOVE' not in arg]
+    else:
+        filtered_args = [re.sub(r'\s*SIP_PYARGREMOVE6\s*', ' ', arg)
+                         for arg in arguments_list if
+                         not ('SIP_PYARGREMOVE' in arg and 'SIP_PYARGREMOVE6' not in arg)]
+
+    # Reassemble the function signature
+    remaining_args = ', '.join(filtered_args)
+    if remaining_args and prefix.strip():
+        prefix += ' '
+    if remaining_args and suffix.strip():
+        suffix = ' ' + suffix
+    return f"{prefix}{remaining_args}{suffix}"
+
+
+def fix_annotations(line):
+    global CONTEXT
+
+    # Get removed params to be able to drop them out of the API doc
+    removed_params = re.findall(r'(\w+)\s+SIP_PYARGREMOVE', line)
+    if CONTEXT.is_qt6:
+        removed_params = re.findall(r'(\w+)\s+SIP_PYARGREMOVE6?', line)
+    for param in removed_params:
+        CONTEXT.skipped_params_remove.append(param)
+        dbg_info(f"caught removed param: {CONTEXT.skipped_params_remove[-1]}")
+
+    _out_params = re.findall(r'(\w+)\s+SIP_OUT', line)
+    for param in _out_params:
+        CONTEXT.skipped_params_out.append(param)
+        dbg_info(f"caught removed param: {CONTEXT.skipped_params_out[-1]}")
+
+    # Printed annotations
+    replacements = {
+        r'//\s*SIP_ABSTRACT\b': '/Abstract/',
+        r'\bSIP_ABSTRACT\b': '/Abstract/',
+        r'\bSIP_ALLOWNONE\b': '/AllowNone/',
+        r'\bSIP_ARRAY\b': '/Array/',
+        r'\bSIP_ARRAYSIZE\b': '/ArraySize/',
+        r'\bSIP_DEPRECATED\b': '/Deprecated/',
+        r'\bSIP_CONSTRAINED\b': '/Constrained/',
+        r'\bSIP_EXTERNAL\b': '/External/',
+        r'\bSIP_FACTORY\b': '/Factory/',
+        r'\bSIP_IN\b': '/In/',
+        r'\bSIP_INOUT\b': '/In,Out/',
+        r'\bSIP_KEEPREFERENCE\b': '/KeepReference/',
+        r'\bSIP_NODEFAULTCTORS\b': '/NoDefaultCtors/',
+        r'\bSIP_OUT\b': '/Out/',
+        r'\bSIP_RELEASEGIL\b': '/ReleaseGIL/',
+        r'\bSIP_HOLDGIL\b': '/HoldGIL/',
+        r'\bSIP_TRANSFER\b': '/Transfer/',
+        r'\bSIP_TRANSFERBACK\b': '/TransferBack/',
+        r'\bSIP_TRANSFERTHIS\b': '/TransferThis/',
+        r'\bSIP_GETWRAPPER\b': '/GetWrapper/',
+        r'SIP_PYNAME\(\s*(\w+)\s*\)': r'/PyName=\1/',
+        r'SIP_TYPEHINT\(\s*([\w\.\s,\[\]]+?)\s*\)': r'/TypeHint="\1"/',
+        r'SIP_VIRTUALERRORHANDLER\(\s*(\w+)\s*\)': r'/VirtualErrorHandler=\1/',
+    }
+
+    if not CONTEXT.is_qt6:
+        replacements[r'SIP_THROW\(\s*([\w\s,]+?)\s*\)'] = r'throw( \1 )'
+    else:
+        # these have no effect (and aren't required) on sip >= 6
+        replacements[r'SIP_THROW\(\s*([\w\s,]+?)\s*\)'] = ''
```

Comment:
```suggestion
    replacements[r'SIP_THROW\(\s*([\w\s,]+?)\s*\)'] = ''
```

if I'm not wrong, this would kill tons of warnings in the build

Reply:
this has no impact on procuded sip files, because we're not on Qt6.
Also it seems we don't have any `SIP_THROW` in Kadas source.

Reply:
this does have an impact, qt6 is a (not ideal) proxy for "recent sip version".
... and in this case it seems that the warnings come from the included qgis files instead (so it's rather something to be patched in the qgis port if anywhere)

Reply:
ok, so we should rather check the version of sip rather than the version of Qt.